test/gcloud/helpers/cds.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+//
+// Provides a global function for clearing a Cloud Datastore database. This is
+// necessary for giving a clean slate to each Cloud Datastore test.
+//
+
+var env = require('process').env;
+var http = require('http');
+
+require('./foam.js');
+
+global.clearCDS = function() {
```

Comment:
Actually I think you have a race condition here.  This file is loading FOAM and so is your other foam.js file.

But the other file is the only one setting the flags.  If this file gets loaded first, FOAM will not be loaded with the gcloud flag.

I think you should remove require('foam.js') from this file, and ensure that the other foam.js helper is listed first int he jasmine.conf file.